### PR TITLE
Regenerate requirements files - dependencies dropping support for python 3.7

### PR DESCRIPTION
**Pull request by eapolinario**
Signed-off-by: Eduardo Apolinario <eapolinario@users.noreply.github.com>

# TL;DR
Some flytekit dependencies are already dropping support for python 3.7, including `numpy` and `pandas`. 

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
In their latest release both pandas (starting on [version 1.4.0](https://github.com/pandas-dev/pandas/releases/tag/v1.4.0)) and numpy (starting on [version 1.22.0](https://github.com/numpy/numpy/releases/tag/v1.22.0)) dropped support for python 3.7, which is supposed to reach EOL only on June-2023 (according to https://www.python.org/dev/peps/pep-0537/#id27).

This PR supersedes https://github.com/flyteorg/flytekit/pull/845.

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Review by wild-endeavor — DISMISSED** [review]

File: .github/workflows/pythonbuild.yml [hunk 1]
```diff
@@ -99,7 +98,6 @@ jobs:
         key: ${{ format('{0}-pip-{1}', runner.os, hashFiles('dev-requirements.txt', format('plugins/{0}/requirements.txt', matrix.plugin-names ))) }}
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip==21.2.4 setuptools wheel
```
Comment (eapolinario) [inline .github/workflows/pythonbuild.yml:102]: `pip` is now installed in the `setup` make target.

**Review by wild-endeavor — APPROVED** [review]
